packages/signers/src/types.ts

```diff
@@ -0,0 +1,26 @@
+import type { Hash, Hex, SignTypedDataParameters } from "viem";
+
+export interface SmartAccountSigner<Inner, AuthParams, UserDetails> {
+  signerType: string;
+  inner: Inner;
+  user: User<UserDetails> | undefined;
+
+  getAddress: () => Promise<Hash>;
+
+  signMessage: (msg: Uint8Array | Hex | string) => Promise<Hex>;
+
+  signTypedData: (params: SignTypedDataParameters) => Promise<Hex>;
+
+  authenticateUser: (params: AuthParams) => Promise<UserDetails>;
+
+  getUserDetails: () => Promise<UserDetails>;
+}
+
+export interface User<UserDetails> {
+  id: string;
+  details: UserDetails;
```

Comment:
let's drop this

Reply:
id + details, or just details?

Reply:
just details, since we have a `getDetails` method